### PR TITLE
Extract methods into the ConnectionInterface.

### PR DESCRIPTION
**Pull request by markstory**
This builds a very minimal interface for connections. I've tried to focus it on only what is needed for fixture interoperability and ConnectionManager functionality. Trying to abstract anything else seemed
foolish at this point.

I added a new `disableConstraints()` method to Connection. This method works like `transactional()` and aims to hide the implementation of if/how foreign keys/constraints are managed in a connection.

Refs #6288

### CODE REVIEW

**Comment by markstory** [comment]
This will require additional changes in the elasticsearch plugin.

File: src/TestSuite/Fixture/FixtureManager.php [hunk 1]
```diff
@@ -284,9 +284,9 @@ protected function _runOperation($fixtures, $operation)
                 $db->logQueries(false);
             }
             $db->transactional(function ($db) use ($fixtures, $operation) {
-                $db->disableForeignKeys();
-                $operation($db, $fixtures);
-                $db->enableForeignKeys();
+                $db->disableConstraints(function ($db) use ($fixtures, $operation) {
+                    $operation($db, $fixtures);
+                });
```
Comment (lorenzo) [inline src/TestSuite/Fixture/FixtureManager.php]: great idea!

**Comment by lorenzo** [comment]
One thing that I recently learn is that `inheritDoc` does not inherit parameter docs :(

**Comment by markstory** [comment]
@lorenzo boourns. I will updatw the docs.

**Pull request merged**